### PR TITLE
[networking_mapping] Add the Ansible wrapper role

### PR DESCRIPTION
**Pull request by pablintino**
As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date

### CODE REVIEW

File: roles/networking_mapper/README.md
Comment (cjeanner) [inline roles/networking_mapper/README.md:1]: please update the README :)

File: roles/networking_mapper/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+# All variables intended for modification should be placed in this file.
+# All variables within this role should have a prefix of "cifmw_networking_mapper"
+cifmw_networking_mapper_basedir: >-
+  {{
+    cifmw_basedir | default(ansible_user_dir ~ '/ci-framework-data')
+  }}
+cifmw_networking_mapper_infra_dir: "/etc/ci/env"
+cifmw_networking_mapper_networking_def_path: "{{ cifmw_networking_mapper_infra_dir }}/networking-definition.yml"
```
Comment (cjeanner) [inline roles/networking_mapper/defaults/main.yml:25]: if you want to keep lines shorter, you may use the `path_join` filter. We should consider this (included myself) when we have long paths.

File: roles/networking_mapper/vars/main.yml
Comment (cjeanner) [inline roles/networking_mapper/vars/main.yml:1]: we can remove this file if empty

**Review by cjeanner — CHANGES_REQUESTED** [review]

File: plugins/action/ci_net_map.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+- This task plugin requires networking facts to be present for every 
+  instance referenced in the Networking Definition.
+- Partial mappings are allowed to allow infrastructure bootstrapping based on the
+  mapping result even without already deployed instances.
+
+options:
+  networking_definition:
+    description:
+    - The Networking Definition as a dictionary.
+    type: str
```
Comment (cescgina) [inline plugins/action/ci_net_map.py:30]: Does this mean that `networking_definition` should be passed as a string containing a dictionary, or the type should be dict?

File: roles/networking_mapper/README.md
Comment (pablintino) [reply roles/networking_mapper/README.md:1]: Done.

**Comment by viroel** [comment]
in a first pass, looks good

File: plugins/action/ci_net_map.py [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+            dict(task_vars["hostvars"]),
+            task_vars["groups"],
+            options=networking_mapper.NetworkingMapperOptions.from_dict(task_args),
+        )
+
+        try:
+            mapping_result = (
+                mapper.map_complete(networking_definition, interfaces_info_dict)
+                if is_complete_map
+                else mapper.map_partial(networking_definition)
```
Comment (lewisdenny) [inline plugins/action/ci_net_map.py:317]: If `interfaces_info` is not provided `else mapper.map_partial(networking_definition)` handles it? I see `interfaces_info` is optional, what is the use case for a partial map?

Comment (pablintino) [reply plugins/action/ci_net_map.py:317]: Yes, it's handled internally properly. That use case is for scenarios where the bootstraping of resources like instances or the networking of the instance requires networking information such as MTU, network ranges, lack or presence of IPv6, etc. For those scenarios the mapper can be called twice, one first call to perform a best effort map without the instances (or part of them only) and a second call with the mac addresses of each instance that will output the entire mapping.

**Review by cjeanner — APPROVED** [review]

**Pull request merged**